content/ko/blog/_posts/2022-05-13-grpc-probes-in-beta.md

```diff
@@ -0,0 +1,171 @@
+
+2018년 8월 21일에 [생성](https://github.com/grpc-ecosystem/grpc-health-probe/commit/2df4478982e95c9a57d5fe3f555667f4365c025d)되었으며, [2018년 9월 19일](https://github.com/grpc-ecosystem/grpc-health-probe/releases/tag/v0.1.0-alpha.1)에 첫 번째 릴리스와 함께 이를 활성화하는데 일반적으로 사용되는 도구가 있다다.
+
+gRPC 앱 상태 확인을 위한 이 접근 방식은 매우 인기있다. `grpc_health_probe`가 있는 [Dockerfile은 3,626개](https://github.com/search?l=Dockerfile&q=grpc_health_probe&type=code), GitHub(작성 시점)의 기본 검색으로 발견된 [yaml 파일은 6,621개](https://github.com/search?l=YAML&q=grpc_health_probe&type=Code) 있다. 이것은 도구의 인기와 이를 기본적으로 지원해야 할 필요성을 잘 나타낸다.
+
+쿠버네티스 v1.23은 gRPC를 사용하여 워크로드 상태를 쿼리하기 위한 알파 품질의 구현을 기본 지원으로 도입했다. 알파 기능이었기 때문에 v1.23 릴리스에서는 기본적으로 비활성화되었다.
+
+## 기능 사용
+
+우리는 다른 프로브와 유사한 방식으로 gRPC 상태를 확인할 수 있도록 구축했으며, 쿠버네티스의 다른 프로브에 익숙하다면 [사용하기 쉬울 것](/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe)이라 믿는다.
+기본적으로 지원되는 프로브는 `grpc_health_probe` 실행 파일과 관련된 해결 방법에 비해 많은 이점이 있다.
+
+기본 gRPC 지원을 사용하면 이미지와 함께 `10MB`의 추가 실행 파일을 다운로드하여 저장할 필요가 없다.
+Exec 프로브는 실행 파일을 실행하기 위해 새 프로세스를 인스턴스화해야 하므로 일반적으로 gRPC 호출보다 느리다.
+또한 파드가 최대 리소스에서 실행 중이고 새 프로세스를 인스턴스화하는데 문제가 있는 경우 검사가 합리적으로 처리되지 않는다.
+
+그러나 몇 가지 제한 사항이 있다. 프로브용 클라이언트 인증서를 구성하는 것은 어렵기 때문에, 클라이언트 인증이 필요한 서비스는 지원하지 않는다. 기본 제공 프로브도 서버 인증서를 확인하지 않고 관련 문제를 무시한다.
+
+또한 기본 제공 검사는 특정 유형의 오류를 무시하도록 구성할 수 없으며 (`grpc_health_probe`는 다른 오류에 대해 다른 종료 코드를 반환함), 단일 프로브에서 여러 서비스 상태 검사를 실행하도록 "연결"할 수 없다.
+
+그러나 이러한 모든 제한 사항은 gRPC에서 꽤 일반적이며 이에 대한 쉬운 해결 방법이 있다.
+
+## 직접 사용
+
+### 클러스터 수준 설정
+
+오늘 이 기능을 사용해 볼 수 있다. 기본 gRPC 프로브를 시도하려면, `GRPCContainerProbe` 기능 게이트를 활성화하여 쿠버네티스 클러스터를 직접 가동할 수 있다. [사용할 수 있는 도구](/docs/tasks/tools/)가 많이 있다.
+
+기능 게이트 `GRPCContainerProbe`는 1.24에서 기본적으로 활성화되어 있으므로, 많은 공급업체에서 이 기능을 즉시 사용할 수 있다.
+따라서 선택한 플랫폼에서 1.24 클러스터를 생성할 수 있다. 일부 공급업체는 1.23 클러스터에서 알파 기능을 사용 할 수 있도록 허용한다.
+
+예를 들어 작성하는 순간, 빠른 테스트를 위해 GKE에서 테스트 클러스터를 가동할 수 있다.
+다른 공급업체도 유사한 기능을 가지고 있을 수 있다. 특히 쿠버네티스 1.24 릴리스 이후 이 블로그 게시물을 읽고 있는 경우에는 더욱 그렇다.
+
+GKE에서 다음 명령어를 사용한다. (참고로 버전은 `1.23`이고 `enable-kubernetes-alpha`가 지정됨).
+
+```shell
+gcloud container clusters create test-grpc \
+    --enable-kubernetes-alpha \
+    --no-enable-autorepair \
+    --no-enable-autoupgrade \
+    --release-channel=rapid \
+    --cluster-version=1.23
+```
+
+또한 클러스터에 접근하기 위해서 `kubectl`을 구성할 필요가 있다.:
+
+```shell
+gcloud container clusters get-credentials test-grpc
+```
+
+### 기능 사용
+
+gRPC 프로브가 작동하는 방식을 테스트하기 위해 파드를 생성해 보겠다. 이 테스트에서는 `agnhost` 이미지를 사용한다.
+이것은 모든 종류의 워크로드 테스트에 사용할 수 있는 k8s 유지 이미지다.
+예를 들어, 두 개의 포트를 노출하는 유용한 [grpc-health-checking](https://github.com/kubernetes/kubernetes/blob/b2c5bd2a278288b5ef19e25bf7413ecb872577a4/test/images/agnhost/README.md#grpc-health-checking) 모듈이 있다. 하나는 상태 확인 서비스를 제공하고 다른 하나는 `make-serving` 및 `make-not-serving` 명령에 반응하는 http 포트다.
+
+다음은 파드 정의의 예다. `grpc-health-checking` 모듈 시작 및 포트 `5000` 및 `8080`을 노출하고 gRPC 준비성 프로브를 구성한다.
```

Comment:
```suggestion
다음은 파드 정의 예시이다. 이 예시는 `grpc-health-checking` 모듈을 시작하고, 포트 `5000` 및 `8080`을 노출하며, gRPC 준비성 프로브를 구성한다.
```

Here is an example pod definition. It starts the `grpc-health-checking` module,
exposes ports `5000` and `8080`, and configures gRPC readiness probe: